### PR TITLE
chore: update winnow to 0.6+ and vecmap-rs to 0.2+

### PR DESCRIPTION
**Pull request by gruebel**
- removed the bugfix suffix of the dependencies for `hcl-edit` to reduce dependabots update frequency. let me know, if you are ok with it. I can also do the same for the other crates or change it back 🙂 
- fixed some `clippy` warnings, nothing troublesome

by the way, great project 💪 saved me for doing some regex juggling 😄

### CODE REVIEW

**Comment by martinohmann** [comment]
Hey @gruebel, thanks for your efforts to migrate to winnow 0.6!

CI is failing for nightly. This is not caused by your change, though and is already broken in `main`. I'm going to fix that and will let you know once this is done so you can rebase the PR.

**Comment by martinohmann** [comment]
@gruebel rebasing onto the latest `main` should get rid of the nightly build failure now.

File: crates/hcl-edit/Cargo.toml [hunk 1]
```diff
@@ -30,14 +30,14 @@ default = []
 perf = ["hcl-primitives/perf"]
 
 [dependencies]
-fnv = "1.0.7"
+fnv = "1.0"
 hcl-primitives = { version = "0.1.2", path = "../hcl-primitives" }
-vecmap-rs = "0.1.12"
-winnow = "0.5.15"
+vecmap-rs = "0.2"
```
Comment (martinohmann) [inline crates/hcl-edit/Cargo.toml:35]: Be sure to also bump this in `crates/hcl-rs/Cargo.toml`

**Review by martinohmann — CHANGES_REQUESTED** [review]

File: crates/hcl-edit/Cargo.toml (hunk 1 above)
Comment (gruebel) [reply crates/hcl-edit/Cargo.toml:35]: yeah, double checked it and I didn't forget to adjust it 😄

**Review by martinohmann — APPROVED** [review]

**Pull request merged**